docs/PROJECT_SPEC.md

```diff
@@ -0,0 +1,80 @@
+# 🎯 Objective
+
+Collaboratively develop a command line interface (cli) to easily convert between (the most common) SBOM formats, CycloneDX and SPDX. 
+
+
+# 🗒️ Primary User Workflows
+
+1. Users can convert SBOMs from SPDX to CycloneDX.*
+2. Users can convert SBOMs from CycloneDX to SPDX.*
+
+*this describes the desired user flow. More nuance will have to be considered when it comes to converting between specific SBOM formats. 
+
+
+# 🗒️ Requirements & Scope
+
+## Supported SBOM Formats
+
+SPDX: 2.2, 2.3, 3.0
+CDX: 1.4, 1.5
+
+## Inputs & Parameters
+
+Inputs
+
+- (implied/positional) Input SBOM (required)
+    - Files must be valid .json document
+    - Files must be valid (CDX || SPDX) SBOM
+    - To start, the cli will only accept one input at a time
+- `-c` Configurations / additional inputs (optional)
+    - Input format and version
+    - Output format and version (e.g. CDX 1.5)
+    - Fail on non-perfect translation: true-false
+    - CDX software identifiers priority (purl\cpe\swid)
+    - User-determined fields: list of key-path and values
+- `-h` provides the help information
+- `-o` output filename (optional)
+    - By default: the output filename is the same as the input filename, appended with “-spdx” or “-cyclonedx” depending on the format output.
+    - A string that becomes the filename as a .json file
+    - ❓ should the input be “output.json” or just “output” (DB prefers the former)
```

Comment:
-o output is optional, we would add `.cdx` or `.spdx` and the `.json` (or other future supported encoding) if not already mentioned in the filename (-o=ori -> ori.cdx.json, -o=ori.cdx -> ori.cdx.json, -o=ori.cdx.json -> ori.cdx.json)

Reply:
will update in next PR